nix/t1/conversion/link-verilator-with-dpi.nix

```diff
@@ -0,0 +1,39 @@
+{ lib
+, stdenv
+, zlib
+}:
+
+{ mainProgram
+, verilatorLib
+, dpiLibs
+}:
+
+assert lib.assertMsg (builtins.typeOf dpiLibs == "list") "dpiLibs should be a list of file path";
+
+stdenv.mkDerivation (rec {
+  name = mainProgram;
+  inherit mainProgram;
+
+  dontUnpack = true;
+
+  propagatedBuildInputs = [ zlib ];
+
+  ccArgs = [
```

Comment:
Try `NIX_CFLAGS_COMPILE`, which will be automatically read and handle in stdenv.

Reply:
Here all args are inputs, not flags. And we are actually performing linking here. I feel a bit wired to pass them as CFLAGS.